oracle/oracle.go

```diff
@@ -197,6 +199,36 @@ func (or *Oracle) AdvanceStateToNextSlot(fullBlock *FullBlock) (uint64, error) {
 	return processedSlot, nil
 }
 
+// Unsubscribes validators that are not active in ethereum. Shares their pending rewards to the pool
+// TODO: there are multiple validator statuses that should be considered as "not active" and should be unsubscribed.
+func (or *Oracle) ValidatorCleanup(blockHeader *v1.BeaconBlockHeader) error {
+	or.mutex.Lock()
+	defer or.mutex.Unlock()
+
+	// Only cleanup if we're past the cleanup slot fork
+	if uint64(blockHeader.Header.Message.Slot) >= MainnetCleanupValidatorsSlot {
+		// Get the latest validator information
+		validatorInfo := or.validators.GetValidators()
+
+		// Iterate over all validators. If two or more  validators exit or get slashed in the same slot,
+		// this cleanup will eventually set both of their pending rewards to 0 and share them among the pool
+		for _, validator := range validatorInfo {
```

Comment:
This iterates the whole set of validators? aka 1 million? Can we avoid that?

Reply:
fixed, we will only iterate over currently subscribed oracle validators